helpers/publicarMovimientoDetalle.go

```diff
@@ -0,0 +1,190 @@
+	query := map[string]string{
+		"MovimientoProcesoExternoId__Id": idMovProcExternoCast,
+	}
+
+	fields := []string{
+		"Detalle",
+	}
+
+	// Para traer todos
+	limit := int64(-1)
+
+	// Nota: Se envían los parametros sortby y order como nil, además de offset como 0 por defecto
+	if result, err := models.GetAllMovimientoDetalle(query, fields, nil, nil, 0, limit); err != nil {
+		logs.Error(err)
+		outputError = errorctrl.Error("ListaRubros -  models.GetAllMovimientoDetalle(query, fields, nil, nil, 0, limit)", err, "404")
+		return []models.CuentasMovimientoProcesoExterno{}, outputError
+	} else {
+		// logs.Debug(fmt.Sprintf("result: %+v", result))
+		if len(result) > 0 {
+			var allDetalleCuentas []models.CuentasMovimientoProcesoExterno
+			allDetalleCuentas = make([]models.CuentasMovimientoProcesoExterno, len(result))
+			for k, detalle := range result {
+				var infoFiltro map[string]interface{}
+				json.Unmarshal([]byte(detalle.(map[string]interface{})["Detalle"].(string)), &infoFiltro)
+				var stringFiltro = make(map[string]interface{})
+				for k, prop := range infoFiltro {
+					if k == "RubroId" || k == "FuenteFinanciamientoId" || k == "ActividadId" {
+						switch prop.(type) {
+						case float64:
+							propCast := fmt.Sprintf("%.0f", prop.(float64))
+							stringFiltro[k] = propCast
+						default:
+							// logs.Debug(reflect.TypeOf(prop))
+							stringFiltro[k] = prop
+						}
+					}
+				}
+				if detalleTemp, err := json.Marshal(stringFiltro); err != nil {
+					logs.Error(err)
+					outputError = errorctrl.Error("ListaRubros -  json.Marshal(detalle)", err, "500")
+					return []models.CuentasMovimientoProcesoExterno{}, outputError
+				} else {
+					// logs.Debug("DETALLE: ", detalle)
+					allDetalleCuentas[k] = models.CuentasMovimientoProcesoExterno{
+						Cuen_Pre:     string(detalleTemp),
+						Mov_Proc_Ext: idMovProcExternoCast,
+					}
+				}
+			}
+
+			detalleCuentasRespuesta = RemoveDuplicateElement(allDetalleCuentas)
+
+			// logs.Debug("RESPUESTA DE CUENTAS: ", detalleCuentasRespuesta)
+		} else {
+			err := "No se encontró ningún registro que coincida"
+			logs.Error(err)
+			outputError = errorctrl.Error("ListaRubros - len(result) > 0", err, "404")
+			return []models.CuentasMovimientoProcesoExterno{}, outputError
+		}
+	}
+
+	return detalleCuentasRespuesta, nil
+}
+
+func RemoveDuplicateElement(addrs []models.CuentasMovimientoProcesoExterno) (aResp []models.CuentasMovimientoProcesoExterno) {
+	result := make([]models.CuentasMovimientoProcesoExterno, 0, len(addrs))
+	temp := map[models.CuentasMovimientoProcesoExterno]struct{}{}
+	for _, item := range addrs {
+		if _, ok := temp[item]; !ok {
+			temp[item] = struct{}{}
+			result = append(result, item)
+		}
+	}
+	return result
+}
+
+func MovimientosDetalleCuentas(movimientosDetalle []models.MovimientoDetalle, idMovProcExterno int) (cuentasMovimientoDetalleRespuesta []models.CuentasMovimientoProcesoExterno, outputError map[string]interface{}) {
```

Comment:
1. Linea muy larga
2. Falta documentar, cambiar el `...` del _suggested change_ para ello
```suggestion
// MovimientosDetalleCuentas ...
func MovimientosDetalleCuentas(movimientosDetalle []models.MovimientoDetalle,
idMovProcExterno int) (cuentasMovimientoDetalleRespuesta []models.CuentasMovimientoProcesoExterno,
outputError map[string]interface{}) {
```